dbt/include/global_project/macros/adapters/show.sql

```diff
@@ -1,22 +1,18 @@
+{#
+    We expect a syntax error if dbt show is invoked both with a --limit flag to show
+    and with a limit predicate embedded in its inline query. No special handling is
+    provided out-of-box.
+#}
 {% macro get_show_sql(compiled_code, sql_header, limit) -%}
-  {%- if sql_header -%}
+  {{ adapter.dispatch('get_show_sql', 'dbt')(compiled_code, sql_header, limit) }}
+{% endmacro %}
+
+{% macro default__get_show_sql(compiled_code, sql_header, limit) %}
+  {%- if sql_header is not none -%}
   {{ sql_header }}
   {%- endif -%}
-  {%- if limit is not none -%}
-  {{ get_limit_subquery_sql(compiled_code, limit) }}
-  {%- else -%}
   {{ compiled_code }}
+  {%- if limit is not none %}
+  limit {{ limit }}
   {%- endif -%}
 {% endmacro %}
-
-{% macro get_limit_subquery_sql(sql, limit) %}
```

Comment:
Even though we're no longer using this directly, we might want to keep it. The fact that it's dispatched suggests that concrete adapters probably implemented this somewhere along the line. They would get an exception if they're referencing this macro.

Along those lines, what happens if an adapter has implemented this macro in order to configure `get_show_sql`? That adapter-level implementation is lost now. I don't have a good answer for how to deal with that (and I'm not sure we can), but are we comfortable with that risk?

Reply:
Oh great points. I'll stick to macro interfaces in the future for these very important callouts.